src/probeinterface/probe.py

```diff
@@ -838,36 +838,75 @@ def from_dict(d: dict) -> "Probe":
 
-        Equivalent to the 'to_dataframe()' pandas function, but without pandas dependency.
+        Similar to the 'to_dataframe()' pandas function, but without pandas dependency.
 
-        Very useful to export/slice/attach to a recording.
+        The intended use is to attach this array to a recording object as a property ("contact vector")
 
         Parameters
         ----------
         complete : bool, default: False
             If True, export complete information about the probe,
-            including contact_plane_axes/si_units/device_channel_indices
+            including contact_plane_axes/si_units/device_channel_indices.
 
-        returns
-        ---------
+        Returns
+        -------
         arr : numpy.array
-            With complex dtype
+            Structured array with the following dtype schema:
+
+            dtype = [
+                ('x', 'float64'),
+                ('y', 'float64'),
+                ('z', 'float64', optional),
+                ('contact_shapes', 'U64'),
+                # Shape parameters
+                ('shape_param_1', 'float64'),
+                ('shape_param_2', 'float64'),
+                                ⋮
+                                ⋮
+                                ⋮
+                variable number of shape parameters
+                ...
+                ('shank_ids', 'U64'),
+                ('contact_ids', 'U64'),
+
+                # The rest is added only if `complete=True`
+                ('device_channel_indices', 'int64', optional),
+                ('si_units', 'U64', optional),
+                ('plane_axis_x_0', 'float64', optional),
+                ('plane_axis_x_1', 'float64', optional),
+                ('plane_axis_y_0', 'float64', optional),
+                ('plane_axis_y_1', 'float64', optional),
+                ('plane_axis_z_0', 'float64', optional),
+                ('plane_axis_z_1', 'float64', optional),
+                # Annotations
+                ('annotation_name_1', 'dtype of annotation', optional),
+                ('annotation_name_2', 'dtype of annotation', optional),                                ⋮
+                                ⋮
+                                ⋮
+                                ⋮
+                variable number of annotations
+                ...
+            ]
         """
 
+        # First define the dtype
         dtype = [("x", "float64"), ("y", "float64")]
         if self.ndim == 3:
             dtype += [("z", "float64")]
+
         dtype += [("contact_shapes", "U64")]
         param_shape = []
         for i, p in enumerate(self.contact_shape_params):
             for k, v in p.items():
                 if k not in param_shape:
                     param_shape.append(k)
         for k in param_shape:
-            dtype += [(k, "float64")]
+            dtype += [
+                (k, "float64")
+            ]  # TODO: I think we should check for the type of the values, what if they are string?
```

Comment:
they must be float, where should we check?

Reply:
Should they, I think i was a dictionary where people could add anything? Maybe that's what I got wrong.

Reply:
no param shape is either the value of the radius for circle contacts or widths/heights for rect/square ;)

Reply:
Good to know. If there are three parameters only then maybe we can avoid the loop and make this explicit as that's the reason of where my confusion arose (plus the docstring I added is wrong), what do you think?